open_instruct/utils.py

```diff
@@ -0,0 +1,338 @@
+                "set True will benefit LLM loading time and RAM consumption."
+            )
+        },
+    )
+    dataset_name: Optional[str] = field(
+        default=None, metadata={"help": "The name of the dataset to use (via the datasets library)."}
+    )
+    dataset_config_name: Optional[str] = field(
+        default=None, metadata={"help": "The configuration name of the dataset to use (via the datasets library)."}
+    )
+    train_file: Optional[str] = field(default=None, metadata={"help": "The input training data file (a json/jsonl file)."})
+    # max_train_samples: Optional[int] = field(
+    #     default=None,
+    #     metadata={
+    #         "help": (
+    #             "For debugging purposes or quicker training, truncate the number of training examples to this "
+    #             "value if set."
+    #         )
+    #     },
+    # )
+    # streaming: bool = field(default=False, metadata={"help": "Enable streaming mode"})
+    # overwrite_cache: bool = field(
+    #     default=False, metadata={"help": "Overwrite the cached training and evaluation sets"}
+    # )
+    preprocessing_num_workers: Optional[int] = field(
+        default=None,
+        metadata={"help": "The number of processes to use for the preprocessing."},
+    )
+    max_seq_length: Optional[int] = field(
+        default=None,
+        metadata={
+            "help": ("The maximum total input sequence length after tokenization. Sequences longer than this will be truncated,")
+        },
+    )
+    overwrite_cache: bool = field(
+        default=False,
+        metadata={"help": "Overwrite the cached training and evaluation sets"}
+    )
+    add_bos: bool = field(
+        default=False,
+        metadata={"help": "Forcibly add bos token to the beginning of the input sequence. Use only when tokenizer does not add bos token by default."},
+    )
+    clip_grad_norm: float = field(
+        default=-1,
+        metadata={"help": "Clip gradient norm. Not compatible with deepspeed (use deepspeed config instead)."},
+    )
+    gradient_accumulation_steps: int = field(
+        default=1,
+        metadata={"help": "Number of updates steps to accumulate before performing a backward/update pass."},
+    )
+    learning_rate: float = field(
+        default=2e-5,
+        metadata={"help": "The initial learning rate for AdamW optimizer."},
+    )
+    logging_steps: Optional[int] = field(
+        default=None,
+        metadata={"help": "Log the training loss and learning rate every logging_steps steps."},
+    )
+    lora_rank: int = field(
+        default=64,
+        metadata={"help": "The rank of lora."},
+    )
+    lora_alpha: float = field(
+        default=16,
+        metadata={"help": "The alpha parameter of lora."},
+    )
+    lora_dropout: float = field(
+        default=0.1,
+        metadata={"help": "The dropout rate of lora modules."},
+    )
+    lr_scheduler_type: str = field(
```

Comment:
Before we had this as a `SchedulerType` arg, not a string, and so the code errors out due to this being a string... unsure of the quickest way to fix.

Reply:
I noticed this, but looking at the docs it made it so string was valid. Double checking.

Reply:
https://huggingface.co/docs/transformers/en/main_classes/optimizer_schedules#transformers.get_scheduler

`name (str or SchedulerType)`

Reply:
cool! seems to work in  my testing now.